Code/Core/Process/Process.cpp

```diff
@@ -183,23 +186,35 @@ bool Process::Spawn( const char * executable,
             }
             else
             {
-                if ( ! CreatePipe( &m_StdOutRead, &m_StdOutWrite, &sa, MEGABYTE ) )
+                if ( ! CreatePipe( &m_StdOutRead, &stdOutWrite, &sa, MEGABYTE ) )
                 {
                     return false;
                 }
-                SetHandleInformation( m_StdOutRead, HANDLE_FLAG_INHERIT, 0 );
+                HANDLE tmp = INVALID_HANDLE_VALUE;
+                DuplicateHandle(GetCurrentProcess(),m_StdOutRead,GetCurrentProcess(),&tmp,0,FALSE,DUPLICATE_SAME_ACCESS);
```

Comment:
It seems like the core of this fix can be kept without the use of DuplicateHandle - i.e. unless I'm missing something, the important part is closing the "other side" of the pipes.  SetHandleInformation can still be used to prevent inheriting the side we keep open.

Re-strucuturing the code a little bit, I think the following would be much clearer. What do you think?
```
    HANDLE stdOutWrite = INVALID_HANDLE_VALUE;
    HANDLE stdErrWrite = INVALID_HANDLE_VALUE;
    HANDLE stdInRead = INVALID_HANDLE_VALUE;

    // Create pipes
    bool ok = true;
    ok &= CreatePipe( &m_StdOutRead, stdOutWrite, &sa, MEGABYTE ) );
    ok &= CreatePipe( &m_StdErrRead, stdErrWrite, &sa, MEGABYTE ) );
    ok &= CreatePipe( &stdInRead, &m_StdInWrite, &sa, MEGABYTE ) );

    // Close the "other" end of the pipes to avoid deadlocks
    if ( stdOutWrite != INVALID_HANDLE_VALUE ) { CloseHandle( stdOutWrite ); }
    if ( stdErrWrite != INVALID_HANDLE_VALUE ) { CloseHandle( stdErrWrite ); }
    if ( stdInRead != INVALID_HANDLE_VALUE ) { CloseHandle( stdInRead ); }

    // Handle failure
    if ( !ok )
    {
        if ( m_StdOutRead != INVALID_HANDLE_VALUE ) { CloseHandle( m_StdOutRead ); }
        if ( m_StdErrRead != INVALID_HANDLE_VALUE ) { CloseHandle( m_StdErrRead ); }
        if ( m_StdInWrite != INVALID_HANDLE_VALUE ) { CloseHandle( m_StdInWrite ); }
        return false;
    }

    // Prevent child inheriting handles
    VERIFY( SetHandleInformation( m_StdOutRead, HANDLE_FLAG_INHERIT, 0 ) );
    VERIFY( SetHandleInformation( m_StdErrRead , HANDLE_FLAG_INHERIT, 0 ) );
    VERIFY( SetHandleInformation( m_StdInWrite, HANDLE_FLAG_INHERIT, 0 ) );
```